guacamole-ext/src/main/java/org/apache/guacamole/net/auth/TokenInjectingUserContext.java

```diff
@@ -83,8 +83,13 @@ public TokenInjectingUserContext(UserContext userContext) {
      * @return
      *     The tokens which should be added to the in-progress call to
      *     connect().
+     *
+     * @throws GuacamoleException
+     *     If the tokens applicable to the given connection cannot be
+     *     generated.
```

Comment:
Are there implications any of the other existing users of this class for adding this exception, here?  Presumably this represents a change to the API, which might bump these changes into 2.0.0 version?

Reply:
There are API compatibility implications, but not ABI compatibility implications. An extension written against the older API will still work as expected against the newer API. From https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.21:

>
> ### 13.4.21. Method and Constructor Throws
>
> Changes to the throws clause of methods or constructors do not break compatibility with pre-existing binaries; these clauses are checked only at compile time.
>

An extension that is updated to build against the new API may need code changes if it invokes its own implementation of this class, but that's it.

This is in line with past Guacamole releases where the version number does not need a full major bump so long as extensions written against the older API will continue to function, even if updating an extension to use the newer API will require code changes. For example, from [the 1.3.0 release notes](https://guacamole.apache.org/releases/1.3.0/#deprecation--compatibility-notes):

>
> # Deprecation / Compatibility notes
>
> Each 1.x release of Apache Guacamole should be compatible with components of older 1.x releases. This compatibility is intended at the Guacamole protocol level and at the extension level, but not necessarily at the API level. This means:
>
> * Extensions from older 1.x releases should still work in binary form, but may need code changes before their source will build against a newer version of guacamole-ext.
> ...